### DISCUSSION TITLE
Creation of `lucky notes` command?

### DISCUSSION THREAD

**Discussion by stephendolan** [luckyframework/lucky#1420] [Feature requests & Ideas]
We were discussing this in Discord the other day, and I think it'd be a cool backlog issue to track on to create a new CLI task called `notes` that jumps in and finds instances of some common "TODO" comments, and then prints them out.

Crystal already kind of does this in **documentation**:
https://crystal-lang.org/reference/conventions/documenting_code.html#flagging-classes-modules-and-methods

I think it'd be neat to copy Rails' implementation for a more command-line based approach:
https://github.com/rails/rails/blob/main/railties/lib/rails/source_annotation_extractor.rb

I figured we could start in a discussion, then transfer to an issue if we agree it's a cool backlog feature :)

**Comment by robacarp** [comment]
@stephendolan does ameba do this? it seems like a great use case for a more general purpose tool like ameba

**Reply by jwoertink** [reply]
I think ameba would be more along the lines of linting your comment tags as opposed to just telling you which tags you've listed. It would be like `use TODO: instead of FIXME:` or something.

**Comment by jwoertink** [comment]
I'm not sure if we'd need a whole task for this or not. I originally thought there was a crystal CLI command that just did it, but I was getting that mixed up with the rails rake task. 

Right now you can easily find them with:

```
grep -Rin "TODO" src/

ack "TODO" src/
```

I think if we did end up adding a task for this, we'd want to make it simple enough that it was super easy to use and remember, but also useful enough that you'd want to use it over just grep. We could definitely make the output a lot prettier. `ack` has a decent output

<img width="996" alt="Screen Shot 2021-02-26 at 8 18 46 AM" src="https://user-images.githubusercontent.com/2391/109325933-620e2680-780b-11eb-877a-040ee3ff187b.png">
<img width="642" alt="Screen Shot 2021-02-26 at 8 19 07 AM" src="https://user-images.githubusercontent.com/2391/109325937-633f5380-780b-11eb-8ff8-fe1ab9d83d77.png">

It would just be tricky to keep it cross-platform, and super fast.
